webbpsf/wfirst.py

```diff
@@ -324,6 +342,23 @@ def _validateConfig(self, **kwargs):
             pass
         super(WFI, self)._validateConfig(**kwargs)
 
+    def toggle_pupil_mask_on(self):
+        self.auto_pupil = False
+        _log.info("Using custom pupil mask")
+        self.pupil = self._masked_pupil_path
+
+    def toggle_pupil_mask_off(self):
+        self.auto_pupil = False
+        _log.info("Using custom pupil mask")
+        self.pupil = self._unmasked_pupil_path
```

Comment:
This part could be simplified into a single function. It clutters up the API to have toggle functions with no parameters!

```
 def toggle_pupil_mask(self, set_pupil_mask_on):
         self.auto_pupil = False
         _log.info("Using custom pupil mask")
         if set_pupil_mask_on: 
             self.pupil = self._masked_pupil_path
        else:
             self.pupil = self._unmasked_pupil_path
```

And there is no need to have a the toggle_pupil_default function, because that can already be accomplished just by directly setting the `.auto_pupil` attribute. I don't see any need to manually reset the self.pupil attribute after that, because that will just happen in `validate_config` whenever `.auto_pupil==True`. 

Alternatively toggling `.auto_pupil` could be incorporated into the `toggle_pupil_mask` function, like so: 

```
def toggle_pupil_mask(self, set_pupil_mask_on):
""" Determine whether to use the pupil mask

     Parameters
     ------------
     set_pupil_mask_on : bool or None
          Set to True or False to force using or not using the cold pupil mask, or to None for the automatic behavior.
"""

     if set_pupil_mask_on is None:
         self.auto_pupil = True
    else:
         self.auto_pupil = False
         _log.info("Using custom pupil mask")
         if set_pupil_mask_on: 
             self.pupil = self._masked_pupil_path
        else:
             self.pupil = self._unmasked_pupil_path
```

Reply:
This looks good. I used your second snippet, thanks!